controllers/network.go

```diff
@@ -282,3 +283,57 @@ func createNetwork(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(network)
 }
+
+// swagger:route PUT /api/networks networks updateNetwork
+//
+// Create a network.
+//
+//			Schemes: https
+//
+//			Security:
+//	  		oauth
+//
+//			Responses:
+//				200: networkBodyResponse
+func updateNetwork(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Content-Type", "application/json")
+
+	var netNew models.Network
+
+	// we decode our body request params
+	err := json.NewDecoder(r.Body).Decode(&netNew)
+	if err != nil {
+		logger.Log(0, r.Header.Get("user"), "error decoding request body: ",
+			err.Error())
+		logic.ReturnErrorResponse(w, r, logic.FormatError(err, "badrequest"))
+		return
+	}
+
+	if netNew.AddressRange == "" && netNew.AddressRange6 == "" {
+		err := errors.New("IPv4 or IPv6 CIDR required")
+		logger.Log(0, r.Header.Get("user"), "failed to create netNew: ",
+			err.Error())
+		logic.ReturnErrorResponse(w, r, logic.FormatError(err, "badrequest"))
+		return
+	}
+
+	netOld, err := logic.GetNetwork(netNew.NetID)
+	if err != nil {
+		logger.Log(0, r.Header.Get("user"), "error fetching network: ",
+			err.Error())
+		logic.ReturnErrorResponse(w, r, logic.FormatError(err, "badrequest"))
+		return
+	}
+	_, _, _, _, _, err = logic.UpdateNetwork(&netOld, &netNew)
```

Comment:
can the update be restricted to only groups and users

Reply:
Limited to `ProSettings` now.